scheduler/strategy/balanced.go

```diff
@@ -0,0 +1,74 @@
+package strategy
+
+import (
+	"sort"
+
+	"github.com/docker/swarm/cluster"
+	"github.com/samalba/dockerclient"
+)
+
+type BalancedPlacementStrategy struct {
+	overcommitRatio int64
+}
+
+func (p *BalancedPlacementStrategy) Initialize() error {
+	return nil
+}
+
+func (p *BalancedPlacementStrategy) PlaceContainer(config *dockerclient.ContainerConfig, nodes []*cluster.Node) (*cluster.Node, error) {
+	scores := balancedScores{}
+
+	for _, node := range nodes {
+		nodeMemory := node.UsableMemory()
+		nodeCpus := node.UsableCpus()
+
+		// Skip nodes that are smaller than the requested resources.
+		if nodeMemory < int64(config.Memory) || nodeCpus < config.CpuShares {
+			continue
+		}
+
+		var cpuScore = (node.ReservedCpus() + config.CpuShares) * 100 / nodeCpus
+		var memoryScore = (node.ReservedMemory() + config.Memory) * 100 / nodeMemory
+		var containerScore = int64(len(node.Containers())) + 1
+
+		var total = cpuScore + memoryScore + containerScore
+
+		if cpuScore > 100 || memoryScore > 100 {
+			continue
+		}
+
+		scores = append(scores, &balancedScore{node: node, score: total})
+	}
+
+	if len(scores) == 0 {
+		return nil, ErrNoResourcesAvailable
+	}
+
+	sort.Sort(scores)
```

Comment:
Instead of re-implementing the wheel here, you can just call `sort.Sort(sort.Reverse(scores))` on a `scores` structure. That should clean up a lot of the boilerplate